backend/app/usecase/shortlink/retriever_test.go

```diff
@@ -110,7 +115,7 @@ func TestRetriever_GetShortLink(t *testing.T) {
 			shortLink, err := retriever.GetShortLink(testCase.alias, testCase.expiringAt)
 
 			if testCase.hasErr {
-				assert.NotEqual(t, nil, err)
+				assert.Equal(t, testCase.err, err)
```

Comment:
In Golang, it's the best practice not to check the equality of error suggested both by Dave Chaney and stackoverflow in unit tests: https://www.reddit.com/r/golang/comments/680s4j/using_custom_error_types_for_testing. If we really want to, let's check error types using `error.As` instead of checking equality. The error string was only designed for human readability.

Reply:
It's very likely that we changed the string for better readability and suddenly many tests start failing.